src/Components/Facilities/Facilities.tsx

```diff
@@ -0,0 +1,424 @@
+import HighlightOffIcon from '@mui/icons-material/HighlightOff';
+import Divider from '@mui/material/Divider';
+import MenuItem from '@mui/material/MenuItem';
+import Menu, { MenuProps } from '@mui/material/Menu';
+import EditIcon from '@mui/icons-material/Edit';
+import { useForm } from 'react-hook-form';
+import deleteImg from "../../assets/Email.png"
+import Avatar from '@mui/material/Avatar';
+import facilityImg from "../../assets/tx-seztx-pool-21834-square.avif"
+import noData from "../../assets/freepik--Character--inject-70.png"
+import MoreVertIcon from '@mui/icons-material/MoreVert';
+import DeleteOutlineOutlinedIcon from '@mui/icons-material/DeleteOutlineOutlined';
+import { AuthContext } from '../../Context/AuthContext';
+import { toast } from 'react-toastify';
+
+
+const style = {
+  position: 'absolute' as 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: 400,
+  bgcolor: 'background.paper',
+  boxShadow: 24,
+  borderRadius: 2
+};
+const StyledTableCell = styled(TableCell)(({ theme }) => ({
+  [`&.${tableCellClasses.head}`]: {
+
+    color: theme.palette.common.black
+  },
+  [`&.${tableCellClasses.body}`]: {
+    fontSize: 14,
+  },
+}));
+
+const StyledTableRow = styled(TableRow)(({ theme }) => ({
+  '&:nth-of-type(odd)': {
+    backgroundColor: theme.palette.action.hover,
+  },
+  // hide last border
+  '&:last-child td, &:last-child th': {
+    border: 0,
+  },
+}));
+export default function Facilities() {
+
+  const [showState, setShowState] = useState();
+
+  const handleClose = () => setShowState(false);
+
+
+
+
+
+
+
+  /////////////////// manual logic ///////////////////
+
+  const navigate = useNavigate()
+  const [pageArray, setPageArray] = useState()
+  const [searchValue, setSearchValue] = useState()
+  const [rows, setRows] = useState()
+  const [itemID, setItemID] = useState(0);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [pagesArray, setPagesArray] = useState([]);
+  const [rowsPerPage, setRowsPerPage] = useState(5);
+  const{baseUrl,reqHeaders}=useContext(AuthContext)
+
+  const [anchorElArray, setAnchorElArray] = useState(Array(rows?.length).fill(null));
+  const handleClick = (event, index) => {
+    const newAnchorElArray = [...anchorElArray];
+    newAnchorElArray[index] = event.currentTarget;
+    setAnchorElArray(newAnchorElArray);
+  };
+
+  const handleCloseAncorEl = (index: number) => {
+    const newAnchorElArray = [...anchorElArray];
+    newAnchorElArray[index] = null;
+    setAnchorElArray(newAnchorElArray);
+  };
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    setValue
+  } = useForm()
+
+  const handleChangePage = (event, newPage) => {
+    setCurrentPage(newPage + 1); 
+    getFacilities(newPage + 1); 
+  };
+  const handleChangeRowsPerPage = (event) => {
+    setRowsPerPage(+event.target.value);
+    setCurrentPage(1); 
+    getFacilities(1); 
+  };
+  const handleShowAdd = () => {
```

Comment:
You need to reset the name to empty string, since if we update a facility and go to add facility, the add-modal will open with the last updated facility instead of being empty.